aten/src/ATen/native/LinearAlgebra.cpp

```diff
@@ -2517,25 +2525,65 @@ void compute_T18_scale_square(
   scalar_t theta
 ) {
   // Scale
-  const auto s = at::max(
-    at::zeros_like(norm),
-    at::ceil(at::log2(norm / theta))
-  ).unsqueeze(-1).unsqueeze(-1).to(at::kLong);
-  const auto pow2s = at::pow(2, s);
-  const auto a_scaled = a / pow2s;
+  // We eventually need to do the matrix multiplication to calculate the result.
+  // For exampe, if we have `norm` equal to [27, 6, 6, 0.05], we will end up to
+  // get `s` as [4, 1, 1, 0], so we can use it to get the result by calculating
+  // matrix[0]^(2^4), matrix[1]^(2^1) and matrix[2]^(2^1) one by one to get the
+  // result, such "one by one calculation" will be quite slow.
+  const auto s = (at::floor(at::log2(norm / theta)) + 1).clamp(/*min=*/0);
```

Comment:
Why changing ceil to floor + 1? Also, we should probably cast s to an integer, as it was done before.